### ISSUE TITLE
iptables support

### ISSUE THREAD

**Issue by ManuGithu** [lkl/linux#398]
Hi All,
I am wondering if iptables are supported in lkl ?
When I try to hijack the iptables command I get the following error message:

/.../lkl-linux/tools/lkl/tests/../bin/lkl-hijack.sh iptables -n -L
iptables v1.6.1: can't initialize iptables table `filter': iptables who? (do you need to insmod?)
Perhaps iptables or your kernel needs to be upgraded.

Furthermore, I do not see how I could configure iptables rules when loading the lkl library.

Any help would be appreciated,
Thanks & regards,
Manu

**Comment by thehajime** [comment]
you may need to add configs to use netfilter (iptables).  here is an example for that.

https://github.com/libos-nuse/lkl-linux-pipe/commit/5655cf413f8ea6381b7c1b6ea7c1d0504a3d3d1e#diff-a9a29ba12b4fa53df67434c22cbd9042

**Comment by ManuGithu** [comment]
Thanks it works ! I can hijack the iptables command.

I am now wondering how can set multiple iptables rules when loading the lkl library...

Any idea ?

Thx & regards

**Comment by tavip** [comment]
@ManuGithu iptables-restore should allow you to apply multiple rules.

**Comment by ManuGithu** [comment]
I have a proxy that I would like to hijack with LKL but I need to set first a number of iptables rules.

Unfortunately, current LKL json configuration file does not offer to configure iptables rules... 
I see a couple of way to do this:
1- extend the LKL api and json configuration file
2- have my proxy to use syscalls to configure iptables rules that will be hijacked by LKL

Am I missing a more simple way to set iptables rules before hijacking a process ?

Thanks & regards

**Comment by tavip** [comment]
No, thats pretty much it. And I think option 1 is better, since its the same amount of work but it will benefit other LKL users as well :)
